### ISSUE TITLE
False positive for super-init-not-called

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#305]
Originally reported by: **Claudiu Popa (BitBucket: [PCManticore](http://bitbucket.org/PCManticore), GitHub: @PCManticore)**

---

Given the following code, pylint fails to understand what cls can represent at a given time and emits a super-init-not-called for the entries after `cycle`.

```
#!python
from itertools import chain, cycle

def test():
    for cls in (cycle, chain, ):
        class Subclass(cls):
            def __init__(self, newarg=None, *args):
                cls.__init__(self, *args)
        Subclass(newarg=1)

```

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/305

**Comment by degustaf** [comment]
This issue should have the control_flow tag.

**Comment by PCManticore** [comment]
It should, but I cannot reproduce it any longer. Probably the inference still can't infer the base class of Subclass, but I'm happy that this error is gone. We can improve the inference later.
